### ISSUE TITLE
[BUG] .NET Native Gatekeeper: ArrayBackedPropertyBag has an explicit default constructor which is unsupported

### ISSUE THREAD

**Issue by bostelk** [Azure/azure-sdk-for-net#43500]
### Library name and version

Azure.Core.dll 1.3800.24.12602

### Describe the bug

In my UWP application, when "static analysis for .NET Native" is enabled, the build will fail. The .net native compiler when run, returns a non-successful exit code `1`. If I review the .NET Native Gatekeeper output I see that there's an error message related to `Azure.Core.ArrayBackedPropertyBag`. I'm not sure how harmful this error is because the application will still run when the option is disabled.

It is not clear that this error is causing the build to be unsuccessful but it's the only error message I can find when I run "ilc.exe --gatekeeper <gkargs.rsp>". For the time being, the static analysis option can be disabled to get a successful build.

**The error**
```Azure.Core : Gatekeeper error GK0018 : GK0018 Value type 'Azure.Core.ArrayBackedPropertyBag' has an explicit default constructor which is unsupported. Use an initialization function instead.```

### Expected behavior

The build is successful when static analysis is enabled.

### Actual behavior

A build error is raised and the build fails.

### Reproduction Steps

You could run "ilc.exe --gatekeeper <gkargs.rsp>" with an appropriate gkargs file. Or create a new UWP project and reference Azure. You should check "Compile with .NET Native tool chain" and "Enable static analysis for .NET Native" options. I can try to get a minimal project together later this week.

### Environment

Microsoft Visual Studio Community 2022
Version 17.6.4
VisualStudio.17.Release/17.6.4+33815.320
Microsoft .NET Framework
Version 4.8.09032

runtime.win10-x64.microsoft.net.native.compiler 2.2.10-rel-29722-00

**Comment by annelo-msft** [comment]
> I can try to get a minimal project together later this week.

Yes, this would be helpful to have to ensure when we create a repro that we are replicating the precise issue you're seeing.  If it is also easy to share a little more about how your UWP app is using Azure.Core (what types is it referencing, e.g.?), that would be helpful context as well.  If you also have a representative gkargs file you're willing to share, that would be helpful as well.  Many thanks!

**Comment by bostelk** [comment]
Hi Anne,

Thank you for the super fast response! Here's a minimal UWP project to reproduce the issue: [GateKeeper.zip](https://github.com/Azure/azure-sdk-for-net/files/15029153/GateKeeper.zip).

**Comment by premiaware** [comment]
I have same issue with Azure.Storage.Blobs version 12.20.0

MSBUILD : error : ILT0038: 'Azure.Core.ArrayBackedPropertyBag`2' is a value type with a default constructor. Value types with default constructors are not currently supported. Consider using an explicit initialization function instead

if I roolback Azure.Storage.Blobs to version 12.19.1 all working fine !!!!

**Comment by premiaware** [comment]
When will this problem be solved? 12.22.9 also does not work!

**Comment by premiaware** [comment]
this problem has been open for more than a year, when will it be fixed? this prevents any publishing of UWP apps in the microsoft store. you can fix it please?

**Comment by premiaware** [comment]
today i have try to update to 12.26.0 but unfortunately the issue persists and it becomes impossible to submit any UWP app to the Miscrosoft Store.
Is it possible to fix this problem permanently?


<img width="785" height="89" alt="Image" src="https://github.com/user-attachments/assets/6d15ddc1-c9a4-4fd3-acf3-afd3f7949b79" />

**Comment by m-redding** [comment]
Fixing in https://github.com/Azure/azure-sdk-for-net/pull/56354

**Comment by premiaware** [comment]
Today i have update to 1.28.0  but the issue is the same, the microsoft store reject the submission.

<img width="709" height="76" alt="Image" src="https://github.com/user-attachments/assets/4de9cc15-543a-4576-a269-d3969be49d92" />

Switch back to 1.19.1 working fine.

**Comment by m-redding** [comment]
@premiaware would you mind opening a new issue with a few more details about the error messages? It might be a similar issue, but it will give the problem more visibility in a new issue rather than here since the specific error (explicit default constructor) was fixed
